### ISSUE TITLE
Inconsistent code generation for fractions

### ISSUE THREAD

**Issue by MaartenBaert** [sympy/sympy#16882]
Sympy currently generates C code for fractions in a very inconsistent way:
```
x = sympy.symbols("x")
sympy.printing.ccode(x/3) # '(1.0/3.0)*x'
sympy.printing.ccode(-x/3) # '-1.0/3.0*x'
sympy.printing.ccode(3/x) # '3/x'
sympy.printing.ccode(-3/x) # '-3/x'
sympy.printing.ccode(1/x) # '1.0/x'
sympy.printing.ccode(-1/x) # '-1/x'
```
Somehow the addition of a minus sign removes the (redundant) parenthesis, and in some cases also causes the coefficient to be printed as an integer rather than a floating point number. The latter  is particularly problematic when `x` is complex, because at least in C++, dividing an integer by a (floating point) complex number produces an error. Is there any way to make the behavior more consistent?

**Comment by asmeurer** [comment]
Redundant parenthese aren't a big deal if they don't change the meaning of the execution. We should definitely be consistent with printing things as ints vs. floats. 

From a technical point of view, in SymPy, `x/3` has a `Rational` in its args, whereas `3/x` has an Integer`, so those take different code paths in the printer. `1/x` is represented as `x**-1`, which is why it is different from the rest. This should be easy to fix but we need to figure out if there are any drawbacks to always printing integers as floats. 

CC @bjodah @moorepants @anpandey

**Comment by bjodah** [comment]
I agree, extra parenthesis isn't a very big deal, inconsistent printing of integer/float could definitely be.

We could add a printer option, maybe `numbers_as_floats` to force the printer to print with decimal points even for numbers which return `True` for `.is_Integer`? But otherwise I think the sensible default is to print sympy `Integer` as C/C++ integers and sympy `Float` as floating point numbers?

Then there's the problem that `/` in C corresponds to `//` in Python (3), perhaps we should represent those as `IntegerDiv` and `FloatDiv` nodes to make things extra clear?
